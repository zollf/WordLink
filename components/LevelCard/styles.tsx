import { StyleSheet } from "react-native";

import AppStyle from "../../styles";

const styles = StyleSheet.create({
  levelCard: {
    width: 60,
    height: 60,
    backgroundColor: AppStyle.grey,
    borderRadius: 5,
    shadowColor: '#000',
    shadowRadius: 4,
    shadowOffset: { width: 2, height: 2},
    shadowOpacity: 0.1,
    margin: 5,
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  },
  text: {
    fontSize: 24,
  }
});

export default styles;