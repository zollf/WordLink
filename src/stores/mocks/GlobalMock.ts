const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
