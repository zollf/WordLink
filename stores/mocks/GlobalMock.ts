const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
