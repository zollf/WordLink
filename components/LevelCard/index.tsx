import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

import styles from './styles';

interface Props {
  number: string,
}

const LevelCard = ({ number }: Props) => {
  return (
    <TouchableOpacity onPress={() => ({})} data-test-id={`card-${number}`} style={styles.levelCard}>
      <Text style={styles.text}>{number}</Text>
    </TouchableOpacity>
  );
};

export default LevelCard;