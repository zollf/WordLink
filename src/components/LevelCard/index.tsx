import React from 'react';
import { TouchableOpacity, Text, View, Image } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useStore } from 'app/src/hooks';

import stars from 'app/src/images/icons/STAR.png';
import AppStyle, { getGradient } from 'app/src/styles';
import styles from './styles';

interface Props {
  level: Level;
  number: string;
  difficulty: string;
}

const LevelCard = ({ level, number, difficulty }: Props) => {
  const { setCurrentPage } = useStore('global');
  const { loadGame } = useStore('gameStore');

  const renderStars = () => {
    const starsArr = [];
    for (let i = 0; i < 3; i++) {
      starsArr.push(<Image key={i} source={stars} style={styles.star} />);
    }
    return starsArr;
  };

  const handleClick = () => {
    loadGame(level);
    setCurrentPage('game');
  };

  return (
    <TouchableOpacity onPress={handleClick} data-test-id="level-card">
      <LinearGradient
        colors={[AppStyle.lightGrey, AppStyle.grey]}
        style={styles.levelCard}
        locations={[0.5, 0.5]}
        start={[-0.1, 0.1]}
      >
        <LinearGradient colors={getGradient(difficulty)} start={[-0.25, 0.25]} style={styles.circle}>
          <Text style={styles.circleText}>{number}</Text>
        </LinearGradient>
        <View style={styles.words}>
          <Text style={styles.text}>{level.game.start}</Text>
          <Text style={styles.text}>{level.game.end}</Text>
        </View>
        <View
          style={[
            styles.quarterCircle,
            {
              backgroundColor: getGradient(difficulty)[1],
            },
          ]}
        />
        <View style={styles.stars}>{renderStars()}</View>
      </LinearGradient>
    </TouchableOpacity>
  );
};

export default LevelCard;
