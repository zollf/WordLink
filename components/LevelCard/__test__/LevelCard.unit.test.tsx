import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from '../../../stores/mocks/GlobalMock';

import LevelCard from '../';

import renderer from 'react-test-renderer';

const number = "1";

const component = (
  <Provider global={GlobalMock}>
    <LevelCard number={number} difficulty="easy"/>
  </Provider>
);

describe('<LevelCard/>', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('function clicks correctly', () => {
    wrapper.root.findByProps({"data-test-id": `card-${number}`}).props.onPress();
    //todo
  });
});