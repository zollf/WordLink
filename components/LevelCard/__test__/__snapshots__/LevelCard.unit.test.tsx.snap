// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelCard/> matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E3E3E3",
      "borderRadius": 5,
      "display": "flex",
      "height": 112.5,
      "justifyContent": "center",
      "margin": 5,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
      "width": 112.5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
      }
    }
  >
    1
  </Text>
</View>
`;
